packages/ui/src/components/NavigationMenu/SNavigationMenu.vue

```diff
@@ -0,0 +1,78 @@
+<script lang="ts">
+export default { name: 'SNavigationMenu' }
+</script>
+
+<script setup lang="ts">
+import { MENU_API_KEY } from '@/components/NavigationMenu/api'
+
+const props = withDefaults(
+  defineProps<{
+    modelValue?: string
+    collapsed?: boolean
+  }>(),
+  {
+    modelValue: '',
+    collapsed: false,
+  },
+)
+
+const emit = defineEmits<(event: 'update:modelValue', value: string) => void>()
+const model = useVModel(props, 'modelValue', emit, { passive: true })
+
+const collapsed = computed(() => props.collapsed)
+
+const api = {
+  active: model,
+  select,
+  collapsed,
+}
+
+function select(value: string) {
+  model.value = value
+}
+
+provide(MENU_API_KEY, api)
```

Comment:
Actually it is a good practice to `provide` APIs wrapped into `readonly()`, so it will prevent some hard-debuggable situations from happening.